src/main/java/org/neo4j/wrapper/LoggingService.java

```diff
@@ -19,35 +19,100 @@
  */
 package org.neo4j.wrapper;
 
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.FileInputStream;
-import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.util.Properties;
+import java.util.logging.LogManager;
 
+/**
+ * The logging service mainly responsible for doing two things:
+ * <ul>
+ * <li>    Reset the property value of {@code java.util.logging.FileHandler.pattern} (which is defined in
+ * {@code windows-wrapper-logging.properties} file) to a absolute path if it is given as a relative path.
+ * The absolute path uses our default working directory {@code %NEO4J_HOME%} as parent.
+ * <li>    Create logging directories if not exists.
+ * </ul>
+ *
+ * The property value of {@code java.util.logging.FileHandler.pattern} uses a special pattern format. More info
+ * about the pattern format could be found in {@link java.util.logging.FileHandler}
+ */
 public class LoggingService
 {
-    private static final String LOGGING_CONFIG_FILE_KEY = "java.util.logging.config.file";
+    static final String LOGGING_CONFIG_FILE_KEY = "java.util.logging.config.file";
+    static final String LOGGING_FILE_NAME_PATTERN_KEY = "java.util.logging.FileHandler.pattern";
+
     private static final String DEFAULT_LOGGING_CONFIG_FILE_PATH = getDefaultLoggingConfigFilePath();
-    private static final String LOGGING_FILE_NAME_PATTERN_KEY = "java.util.logging.FileHandler.pattern";
+    private String winWrapperLogNamePattern;
 
-    public static void initLogger() throws FileNotFoundException, IOException
+    public LoggingService()
     {
-        // Load all the properties from logging.properties
-        String logConfigFileName = System.getProperty( LOGGING_CONFIG_FILE_KEY, DEFAULT_LOGGING_CONFIG_FILE_PATH );
-        Properties logProperties = new Properties();
-        logProperties.load( new FileInputStream( new File( logConfigFileName ) ) );
+    }
 
-        // Find the logging directory and create the directory if not exists.
-        String logFileNamePattern = logProperties.getProperty( LOGGING_FILE_NAME_PATTERN_KEY );
-        File logDir = getLogDir( logFileNamePattern );
+    // only for testing
+    LoggingService( String pattern )
+    {
+        this.winWrapperLogNamePattern = pattern;
+    }
+
+    public void initLogger() throws IOException
+    {
+        LogManager logManager = LogManager.getLogManager();
+        winWrapperLogNamePattern = logManager.getProperty( LOGGING_FILE_NAME_PATTERN_KEY );
+
+        // Reset the pattern property if is relative
```

Comment:
This comment does not really help understand what is happening. While i can see what "reset the pattern" means from the corresponding method's javadoc, i am missing why it is necessary to do so if the pattern is relative